src/hdmf/data_utils.py

```diff
@@ -127,6 +128,19 @@ def maxshape(self):
         raise NotImplementedError("maxshape not implemented for derived class")
 
 
+# TODO: remove this function
+def _temporary_prod(iterable):
+    """
+    We want to use math.prod but it is only available in Python >= 3.8 versions.
+
+    Once 3.7 support is dropped, this function can be removed
+    """
+    result = 1
+    for x in iterable:
+        result *= x
+    return result
```

Comment:
Should this function be be removed?

Reply:
Only once Python 3.7 is no longer supported by HDMF (still the lower bound of the CI right now)